pkg/sessionctx/variable/varsutil.go

```diff
@@ -587,3 +589,24 @@ func ParseAnalyzeSkipColumnTypes(val string) map[string]struct{} {
 	}
 	return skipTypes
 }
+
+func parseSchemaCacheSize(s *SessionVars, normalizedValue string, originalValue string) (byteSize uint64, normalizedStr string, err error) {
+	defer func() {
+		if err == nil && byteSize > 0 && byteSize < (512<<20) {
+			s.StmtCtx.AppendWarning(ErrTruncatedWrongValue.FastGenByArgs(TiDBSchemaCacheSize, originalValue))
+			byteSize = 512 << 20
+			normalizedStr = "512MB"
+		}
+		if err == nil && byteSize > math.MaxInt64 {
+			s.StmtCtx.AppendWarning(ErrTruncatedWrongValue.FastGenByArgs(TiDBSchemaCacheSize, originalValue))
+			byteSize = math.MaxInt64
+			normalizedStr = strconv.Itoa(math.MaxInt64)
+		}
+	}()
+	res, err := units.RAMInBytes(normalizedValue)
+	if err != nil {
+		return 0, "", ErrTruncatedWrongValue.GenWithStackByArgs(TiDBSchemaCacheSize, originalValue)
+	}
+	bt := uint64(res)
```

Comment:
the type of `res` is int64. I'm not sure if line 600 will really happen. Please also add a unit test to cover it.

Reply:
The `RAMInBytes` func will use `strconv.ParseFloat` func to parse the size, it may return negative numbers(res) and then uint64(res) will be larger than math.MaxInt64, so we check it.
There is a ut in `pkg/sessionctx/variable/sysvar_test.go` and we test `math.MaxInt64+1`